### ISSUE TITLE
Inconsistent output when using contiguous

### ISSUE THREAD

**Issue by Lmy0217** [pytorch/pytorch#72512]
### 🐛 Describe the bug

Inconsistent output when using contiguous.

## Minimal example
```Python
import torch
import torch.nn as nn
import torch.nn.functional as F


class LayerNorm2d(nn.LayerNorm):

    def __init__(self, normalized_shape, eps=1e-6):
        super().__init__(normalized_shape, eps=eps)

    def forward(self, x) -> torch.Tensor:
        return F.layer_norm(x.permute(0, 2, 3, 1), self.normalized_shape, self.weight, self.bias, self.eps).permute(0, 3, 1, 2)


if __name__ == '__main__':
    torch.use_deterministic_algorithms(True)
    torch.backends.cudnn.deterministic = True
    torch.backends.cudnn.benchmark = False

    device = torch.device('cuda')
    x = torch.ones(4, 3, 224, 224, device=device, requires_grad=True)

    conv1 = nn.Conv2d(3, 96, kernel_size=1).to(device)
    ln = LayerNorm2d(96).to(device)
    conv2 = nn.Conv2d(96, 96, kernel_size=3, groups=3).to(device)

    x = ln(conv1(x))

    o1 = conv2(x)
    o2 = conv2(x.contiguous())

    diff = o1 - o2
    print(torch.min(diff), torch.max(diff))
```

I get non-zero outputs:
```
tensor(-4.7684e-07, device='cuda:0', grad_fn=<MinBackward1>) tensor(4.4703e-07, device='cuda:0', grad_fn=<MaxBackward1>)
```

### Versions

PyTorch version: 1.10.1+cu113
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 企业版
GCC version: Could not collect
Clang version: Could not collect
CMake version: version 3.19.2
Libc version: N/A

Python version: 3.8.6 (tags/v3.8.6:db45529, Sep 23 2020, 15:52:53) [MSC v.1927 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.19041-SP0
Is CUDA available: True
CUDA runtime version: 11.0.221
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 2080 Ti
Nvidia driver version: 471.41
cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.0\bin\cudnn_ops_train64_8.dll
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] adabelief-pytorch==0.2.1
[pip3] nfnets-pytorch==0.1.3
[pip3] numpy==1.21.4
[pip3] robust-loss-pytorch==0.0.2
[pip3] torch==1.10.1+cu113
[pip3] torch-dct==0.1.5
[pip3] torchaudio==0.10.1+cu113
[pip3] torchio==0.18.70
[pip3] torchstat==0.0.7
[pip3] torchsummary==1.5.1
[pip3] torchvision==0.11.2+cu113
[conda] Could not collect

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @ngimel

**Comment by albanD** [comment]
Hi,

For single precision number 1e-7 is the same as 0. See this note for more details: https://pytorch.org/docs/master/notes/numerical_accuracy.html

Closing as expected behavior.
